### ISSUE TITLE
Dashboard redirect loop after update from v2.0 to v2.1

### ISSUE THREAD

**Issue by schwebke** [containous/traefik#6041]
### Do you want to request a *feature* or report a *bug*?

Bug

#### my Setup
* LoadBalancer including SSL termination and reverse proxy to k8s nodeport
* traefik as k8s service on nodeport
* LoadBalancer IP (range) configured as forwardedheaders.trustedips for entrypoint

#### Update from 2.0.6 to 2.1.1
* added CRD, adapted ClusterRole
* switched from insecure API/dashboard to api@internal / kind TraefikService, still using host based rule

### What did you expect to see?

traefik api, traefik dashboard and other k8s services routed via traefik continue working

### What did you see instead?

API and other services work fine.

I can no longer access traefik dashboard, however. I see the following redirect loop instead:
https://traefik.example.org/dashboard/ ->
http://traefik.example.org/dashboard/dashboard/ ->
https://traefik.example.org/dashboard/dashboard/ ->
http://traefik.example.org/dashboard/dashboard/dashboard/ ->
https://traefik.example.org/dashboard/dashboard/dashboard/ ->
...

For http to https I have an intentional redirect.
Problem is not serving the dashboard in the first place -- I think the first redirect from https://.../dashboard/ (which arrives as http, as SSL is terminated by LB outside the k8c cluster) to http://.../dashboard/dashboard/ is causing the error and this redirect was introduced from 2.0 to 2.1.

traefik api works fine:
GET https://traefik.example.org/api/version
->
```
{
    "Version": "2.1.1",
    "Codename": "cantal"
}
```

All other services works fine, too. Output of echo service (showing forwarded LoadBalancer headers):
```
CLIENT VALUES:
client_address=('10.244.3.17', 53630) (10.244.3.17)
command=GET
path=/echo/
real path=/echo/
query=
request_version=HTTP/1.1

SERVER VALUES:
server_version=BaseHTTP/0.6
sys_version=Python/3.5.0
protocol_version=HTTP/1.0

HEADERS RECEIVED:
Accept=text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Encoding=gzip, deflate, br
Accept-Language=de-DE,en-US;q=0.7,en;q=0.3
Dnt=1
Host=example.org
Te=trailers
Upgrade-Insecure-Requests=1
User-Agent=Mozilla/5.0 (X11; Linux x86_64; rv:68.0) Gecko/20100101 Firefox/68.0
X-Forwarded-For=..., 10.244.4.190
X-Forwarded-Host=example.org
X-Forwarded-Port=443
X-Forwarded-Proto=https
X-Forwarded-Server=traefik-76b56566c6-pg672
X-Real-Ip=10.244.4.190
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
2.1.1
```

#### kubernetes, with the following static config
```yaml
kind: Deployment
apiVersion: apps/v1
metadata:
  namespace: default
  name: traefik
  labels:
    app: traefik

spec:
  replicas: 1
  selector:
    matchLabels:
      app: traefik
  template:
    metadata:
      labels:
        app: traefik
    spec:
      serviceAccountName: traefik-ingress-controller
      containers:
        - name: traefik
          image: traefik:v2.1.1
          args:
            - --api.dashboard=true
            - --accesslog
            - --entrypoints.web.Address=:8000
            - --entrypoints.web.forwardedheaders.trustedips=10.0.0.0/8
            - --providers.kubernetescrd
            - --ping
            - --ping.entrypoint=web
          ports:
            - name: admin
              containerPort: 8080
```

#### node port, entry point
```yaml
apiVersion: v1
kind: Service
metadata:
  name: traefik
  namespace: default

spec:
  type: NodePort
  ports:
    - protocol: TCP
      name: web
      port: 8000
      nodePort: 30000
  selector:
    app: traefik
```

#### traefik dashboard and api
```yaml
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: basic-auth
  namespace: default
spec:
  basicAuth:
    secret: basic-auth-secret
---
apiVersion: v1
kind: Secret
metadata:
  name: basic-auth-secret
  namespace: default
data:
  users: ...
---
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  name: traefik-dashboard-ingressroute
  namespace: default
spec:
  entryPoints:
    - web
  routes:
  - match: Host(`traefik.example.org`)
    kind: Rule
    middlewares:
    - name: basic-auth
    services:
    - name: api@internal
      kind: TraefikService
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
(paste your output here)
```

**Comment by schwebke** [comment]
closing, issue disappeared using `api@internal/kind: TraefikService` instead of `api.insecure` / port based service and clearing browser caches.

**Comment by luolizhi** [comment]
same problem, @schwebke could you paste you `TraefikService` file, thank you in advance.

**Comment by luolizhi** [comment]
thanks @schwebke , i am the same question, resovled my problem.

**Comment by robinwangrubin** [comment]
So..... What should i do to solve this problem?


Here is how i start my Traefik-v2. And i cannot access Dashboard.

```
docker service create \
	--name traefik-v2 \
	--mode global \
	--network docker-proxy --network web \
	--publish mode=host,target=80,published=80,protocol=tcp \
	--publish mode=host,target=443,published=443,protocol=tcp \
	--constraint 'node.role == worker' \
	--label 'traefik.docker.network=web' \
	--label 'traefik.enable=true' \
	--label 'traefik.http.routers.traefik-v2-https.rule=Host(`traefik-v2.xxxx-dev.com`)' \
	--label 'traefik.http.services.traefik-v2.loadbalancer.server.port=8080' \
	--label 'traefik.port=8080' \
	--mount type=bind,source=/usr/share/zoneinfo,destination=/usr/share/zoneinfo,ro=1 \
	traefik:v2.1.1 \
	--providers.docker=true \
	--providers.docker.endpoint=tcp://docker-proxy:2375 \
	--providers.docker.swarmMode=true \
	--providers.docker.swarmModeRefreshSeconds=10s \
	--providers.docker.exposedByDefault=false \
	--entryPoints.web.address=:80 --entryPoints.websecure.address=:443 \
	--log=true --log.level=INFO --api.insecure=true 
```

**Comment by ldez** [comment]
https://community.containo.us/t/serving-traefiks-internal-dashboard-behind-traefik-itself/3457/7?u=ldez

**Comment by traefiker** [comment]
Closed by #6078.
